sdk/logical/storage.go

```diff
@@ -220,11 +220,7 @@ func ClearViewWithLogging(ctx context.Context, view ClearableView, logger hclog.
 		logger = hclog.NewNullLogger()
 	}
 
-	if paginated, ok := view.(ClearableView); ok {
-		return ClearViewWithPagination(ctx, paginated, logger)
-	}
-
-	return ClearViewWithoutPagination(ctx, view, logger)
```

Comment:
@cipherboy `ClearViewWithoutPagination` was never used with this code, so I removed it completely. I do not know if that is the intended outcome, or if the check here was just wrong.

Reply:
`view` is already of type `ClearableView` (as per method signature), so the cast here will always succeed and `ClearViewWithoutPagination` will never be called.

Reply:
Ohh I think I got what you mean. A third party uses the function directly in their code via the sdk import. Gotcha. I'll revert the deletion in that case.